features/taxonomy/managing_taxons/deleting_taxon.feature

```diff
@@ -32,6 +32,14 @@ Feature: Deleting a taxon
         When I try to delete taxon named "Caps"
         Then I should be notified that I cannot delete a menu taxon of any channel
 
+    @ui @javascript @api
+    Scenario: Deleting a taxon that is a main taxon of a product
+        And the store classifies its products as "T-Shirts"
+        And the store has a product "T-Shirts PHP"
+        And this product has a main taxon "T-Shirts"
+        When I delete taxon named "T-Shirts"
+        Then the taxon named "T-Shirts" should no longer exist in the registry
+
```

Comment:
Maybe let's add what happens with the product 🤔?

Reply:
Good catch!